spyder_vim/vim_widget.py

```diff
@@ -250,13 +254,13 @@ class VimLineEdit(QLineEdit):
     def keyPressEvent(self, event):
         if event.key() == Qt.Key_Escape:
             self.clear()
+        elif event.key() == Qt.Key_Backspace:
+            self.setText(self.text() + "\b")
         else:
             QLineEdit.keyPressEvent(self, event)
 
     def focusInEvent(self, event):
-        QLineEdit.focusInEvent(self, event)
-        self.parent().update_vim_cursor()
-        self.clear()
+        self.setText("h")
```

Comment:
When the vim plugin is not focused, it is in insert mode. When leaving insert mode, the vim cursor should move to the left as if h was pressed.
This sets the vim plugin's commandline text to "h" which triggers `VimWidget.on_text_changed(self, text)` so `VimKeys("h", 1)` is called.
Also, I shouldn't have removed `self.parent().update_vim_cursor()` or `QLineEdit.focusInEvent(self, event)` though. I'll put them back in.